components/script/dom/globalscope.rs

```diff
@@ -3244,6 +3240,18 @@ impl GlobalScope {
             .expect("Failed to queue update gamepad state task.");
     }
 
+    /// <https://www.w3.org/TR/gamepad/#dfn-gamepad-user-gesture>
+    pub fn contains_user_gesture(&self, update_type: GamepadUpdateType) -> bool {
```

Comment:
Is clippy not complaining about not using `self`? 

I think this should be a function(not method), inside `gamepad`. Because now it reads as if it's the global that "contains a user gesture" or not. 

also that way you only need to import that function, and the constants are used only in `gamepad`.